.github/workflows/semantic.yml

```diff
@@ -0,0 +1,33 @@
+# Check that the commit messages are between 20 and 150 chars,
+# and that they follow the rule
+# <type> (<scope>): <subject>
+# type: docs, feat, fix, refactor, style or test
+# scope (optional): any extra info, (like DMS or whatever)
+
+name: 'Commit Message Check'
+on: pull_request
+
+jobs:
+  check-commit-message:
+    name: Check Commit Message
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Commit Format
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^(docs|feat|fix|refactor|style|test)( ?\(.*\))?: .+$'
+          excludeDescription: 'true' # optional: this excludes the description body of a pull request
+          excludeTitle: 'true' # optional: this excludes the title of a pull request
+          checkAllCommitMessages: 'true' # optional: this checks all commits associated with a pull request
+          accessToken: ${{ secrets.GITHUB_TOKEN }} # github access token is only required if checkAllCommitMessages is true
+          flags: 'gm'
+          error: 'Your commit has to follow the format "<type>(<scope>): <subject>"".'
```

Comment:
Since the type value is strictly limited, it may be worth noting the list of possible values here? For example:
`"<type (docs, feat, fix, refactor, style, test)>(<scope>): <subject>"`

Or check the type separately and report that it does not match the possible values.

Reply:
I can add it in the doc. Making a different job for each value is a waste of resources